db/crud_test.go

```diff
@@ -956,3 +956,219 @@ func BenchmarkHandleRevDelta(b *testing.B) {
+	rev3, err := db.DeleteDoc("camera", rev2)
+	bodyBytes, removed, err = db.get1xRevFromDoc(doc2, rev3, true)
+	assert.False(t, removed)
+	assert.Error(t, err, "It should throw 404 missing error")
+	assert.Nil(t, bodyBytes, "Document body bytes should be empty")
+
+	// Get the revision details from doc2 and rev2 after rev2 is deleted
+	bodyBytes, removed, err = db.get1xRevFromDoc(doc2, "", true)
+	assert.False(t, removed)
+	assert.NoError(t, err, "It should not throw any error")
+	assert.NotNil(t, bodyBytes, "Document body bytes should be received")
+	assert.NoError(t, response.Unmarshal(bodyBytes))
+}
+
+func TestGet1xRevFromDoc(t *testing.T) {
+	testBucket := base.GetTestBucket(t)
+	defer testBucket.Close()
+	bucket := testBucket.Bucket
+
+	context, err := NewDatabaseContext("db", bucket, false, DatabaseContextOptions{})
+	assert.NoError(t, err, "Couldn't create context for database 'db'")
+	defer context.Close()
+	db, err := CreateDatabase(context)
+	assert.NoError(t, err, "Couldn't create database 'db'")
+
+	// Create the first revision of the document
+	docId := "356779a9a1696714480f57fa3fb66d4c"
+	payload := `{"city":"Los Angeles"}`
+	doc, rev1, err := db.PutExistingRevWithBody(docId, unjson(payload), []string{"1-a"}, false)
+	assert.NoError(t, err, "Couldn't create document")
+	assert.NotEmpty(t, doc, "Document shouldn't be empty")
+	assert.Equal(t, "1-a", rev1, "Provided input revision ID should be returned")
+
+	// Get the 1x revision with rev1
+	bodyBytes, removed, err := db.get1xRevFromDoc(doc, rev1, true)
+	assert.NotEmpty(t, bodyBytes, "Document body bytes should be returned")
+	assert.False(t, removed, "This shouldn't be a removed document")
+	var response = Body{}
+	assert.NoError(t, response.Unmarshal(bodyBytes))
+	assert.Equal(t, docId, response[BodyId])
+	assert.Equal(t, "1-a", response[BodyRev])
+	assert.Equal(t, "Los Angeles", response["city"])
+
+	// Create the second revision of the document
+	payload = `{"city":"Hollywood"}`
+	doc, rev2, err := db.PutExistingRevWithBody(docId, unjson(payload), []string{"2-a", "1-a"}, false)
+	assert.NoError(t, err, "Couldn't create document")
+	assert.NotEmpty(t, doc, "Document shouldn't be empty")
+	assert.Equal(t, "2-a", rev2, "Provided input revision ID should be returned")
+
+	// Get the 1x revision with rev2
+	bodyBytes, removed, err = db.get1xRevFromDoc(doc, rev2, true)
+	assert.NotEmpty(t, bodyBytes, "Document body bytes should be returned")
+	assert.False(t, removed, "This shouldn't be a removed document")
+	assert.NoError(t, response.Unmarshal(bodyBytes))
+	assert.Equal(t, docId, response[BodyId])
+	assert.Equal(t, "2-a", response[BodyRev])
+	assert.Equal(t, "Hollywood", response["city"])
+
+	// Get body bytes from doc with unknown revision id; it simulates the error scenario.
+	// A 404 missing error should be thrown when trying get the body bytes of the document
+	// which doesn't exists in the database. The revision "3-a" doesn't exists in database.
```

Comment:
```suggestion
	// which doesn't exists in the database. The revision "3-a" doesn't exists in the revision tree
```

Reply:
Comment has been modified and changes are available in latest revision.